rs/boundary_node/ic_boundary/src/tls.rs

```diff
@@ -249,9 +247,12 @@ impl Provision for Provisioner {
         self.token_owner.set(None).await;
 
         warn!("TLS: Certificate for {name} successfully provisioned");
+
+        let key_pair_pem = key_pair.serialize_pem();
+        key_pair.zeroize();
         Ok(ProvisionResult::Issued(TLSCert(
-            cert_chain_pem,                   // Certificate Chain
-            cert.serialize_private_key_pem(), // Private Key
+            cert_chain_pem, // Certificate Chain
+            key_pair_pem,   // Private Key
```

Comment:
The "Private Key" comment/description now seems a bit confusing based on the called method `key_pair.serialize_pem()`, which "Serializes the key pair (including the private key) in PKCS#8 format in PEM".  The previously called `cert.serialize_private_key_pem()` "Serializes the private key in PEM format", but digging into what is called under the hood, both end up encoding the response to `key_pair.serialize_der()`, which "Serializes the key pair (including the private key) in PKCS#8 format in DER". So unless I missed something, I think the correct comment would be `// Key Pair`?

Reply:
I also came to the conclusion that `cert.serialize_private_key_pem()` and `key_pair.serialize_pem()` eventually do the same thing. Good point regarding the comment: I updated it as suggested.